subsys/bluetooth/controller/ll_sw/isoal.c

```diff
@@ -1700,6 +1941,41 @@ static isoal_status_t isoal_tx_framed_produce(struct isoal_source *source,
 	return err;
 }
 
+/**
+ * @brief  Handle preparation of the given source before commencing TX on the
+ *         specified event (only for framed sources)
+ * @param  source_hdl   Handle of source to prepare
+ * @param  event_count  Event number source should be prepared for
+ * @return              Status of operation
+ */
+static isoal_status_t isoal_tx_framed_event_prepare_handle(isoal_source_handle_t source_hdl,
+							   uint64_t event_count)
+{
+	struct isoal_source_session *session;
+	struct isoal_pdu_production *pp;
+	struct isoal_source *source;
+	uint64_t last_event_payload;
+	isoal_status_t err;
+
+	err = ISOAL_STATUS_OK;
+
+	source = &isoal_global.source_state[source_hdl];
+	session = &source->session;
+	pp = &source->pdu_production;
+	last_event_payload = (session->burst_number * (event_count + 1)) - 1;
```

Comment:
Any risk over overflow here?

Reply:
This should not overflow 8-bits BN x 39-bits event count. Once again the 64-bits is for convenience.